internal/controllers/machine/machine_controller_noderef.go

```diff
@@ -262,3 +267,28 @@ func (r *Reconciler) getNode(ctx context.Context, c client.Reader, providerID *n
 
 	return &nodeList.Items[0], nil
 }
+
+func (r *Reconciler) reconcileNodeTaints(ctx context.Context, remoteClient client.Client, node *corev1.Node) error {
+	log := ctrl.LoggerFrom(ctx)
+	taints := []corev1.Taint{}
+	patchHelper, err := patch.NewHelper(node, remoteClient)
+	if err != nil {
+		return errors.Wrap(err, "failed to create patch helper")
+	}
+	nodeHasUninitializedTaint := false
+	for _, taint := range node.Spec.Taints {
+		if taint.MatchTaint(&clusterv1.NodeUninitializedTaint) {
+			nodeHasUninitializedTaint = true
+			continue
+		}
+		taints = append(taints, taint)
+	}
+	if nodeHasUninitializedTaint {
+		node.Spec.Taints = taints
+		if err := patchHelper.Patch(ctx, node); err != nil {
+			log.V(2).Info("Failed patch node to set taints", "err", err, "node name", node.Name)
```

Comment:
Q: Is it necessary to log + return the error here?

I would assume wrapping and returning the error should be enough